stack/ArrayStack.java

```diff
@@ -5,16 +5,14 @@
  * 
  * @author Harry Willis
  */
-
-public class ArrayStack<T extends Comparable <T>> implements Stack<T> {
+public class ArrayStack<T extends Object> implements Stack<T> {

```

Comment:
Why does T extend Object here? This new to me :)

Reply:
This is for the search function. It ensures that whatever type is being used for the stack is a subclass of Object and thus has an equals function, which I use in search().